### PR TITLE
Azure Log Errors Alert

### PR DESCRIPTION
**Pull request by jherrflexion**
## Issue

#1395

### CODE REVIEW

File: operations/template/alert.tf [hunk 1]
```diff
@@ -120,3 +120,55 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "database_token_expired_a
...
+  data_source_id = azurerm_linux_web_app.api.id
+  description    = "Alert when total errors cross threshold"
+  enabled        = true
+
+  query = <<-QUERY
+      AppServiceConsoleLogs
+      | where Level == "Error"
+      and TimeGenerated >= ago(30m)
+      and TimeGenerated <= now()
+      | summarize count()
```
Comment (halprin) [inline operations/template/alert.tf:144]: I think the `Level` is not our level in this case.  Looking at `operations/template/logs.tf` and the `azurerm_log_analytics_query_pack_query` resources, we need to dig into the `ResultDescription`.  Taking inspiration from that, I think the query should be something like...

```suggestion
      AppServiceConsoleLogs
      | project JsonResult = parse_json(ResultDescription) | evaluate bag_unpack(JsonResult)
      | where level == 'ERROR'
      and @timestamp >= ago(30m)
      and @timestamp <= now()
      | summarize count()
```

Also, do we need the timestamp checks given that right below the query there's a `time_window`?

File: operations/template/alert.tf [hunk 2]
```diff
@@ -120,3 +120,55 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "database_token_expired_a
...
+    QUERY
+
+  severity                = 3
+  frequency               = 10
+  time_window             = 30
+  auto_mitigation_enabled = true
+
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 1
```
[the remark this answers is not in the document]
Comment (halprin) [reply operations/template/alert.tf:154]: Stop it, you crazy bot.  Not _everything_ needs to be parameterized.  lol

File: operations/template/alert.tf [hunk 3]
```diff
@@ -120,3 +120,55 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "database_token_expired_a
...
+
+resource "azurerm_monitor_scheduled_query_rules_alert" "ti-log-errors-alert" {
+  count               = local.non_pr_environment ? 1 : 0
+  name                = "cdcti-${var.environment}-log-errors-alert"
+  location            = data.azurerm_resource_group.group.location
+  resource_group_name = data.azurerm_resource_group.group.name
+
+  action {
+    action_group  = [azurerm_monitor_action_group.notify_slack_email[count.index].id]
+    email_subject = "TI log errors detected!"
```
Comment (halprin) [inline operations/template/alert.tf:132]: I believe the e-mail subject is easily seen in Slack.  To help differentiate between production and any of the other environments, I recommend putting the environment in the subject line.  Maybe something like...?

```suggestion
    email_subject = "${var.environment}: TI log errors detected!"
```

File: operations/template/alert.tf [hunk 4]
```diff
@@ -120,3 +120,55 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "database_token_expired_a
...
+  }
+
+  data_source_id = azurerm_linux_web_app.api.id
+  description    = "Alert when total errors cross threshold"
+  enabled        = true
+
+  query = <<-QUERY
+      AppServiceConsoleLogs
+      | where Level == "Error"
```
Comment (basiliskus) [inline operations/template/alert.tf:141]: The actual `Level` that you need to filter with is not this one. This one will always return `Informational` AFAIK. You can get the correct Level like this:
```
AppServiceConsoleLogs
| project JsonResult = parse_json(ResultDescription)
| evaluate bag_unpack(JsonResult)
| where level == 'ERROR'
```

File: operations/template/alert.tf (hunk 1 above)
Comment (jherrflexion) [reply operations/template/alert.tf]: Great question, what do you think @saquino0827 ?

Comment (jherrflexion) [reply operations/template/alert.tf]: Imo, we need both. 

The timestamp is the range of the query and the time_window is how frequently it is ran. If anything, we could refactor out the 30 so there are no gaps between the range and frequency.

**Review by halprin — APPROVED** [review]
Assuming this works, I approve.

**Comment by jherrflexion** [comment]
Looks like an internal merge broke this. Going to create a new one.

**Pull request closed without merging**